pypesto/result.py

```diff
@@ -24,7 +23,7 @@ def __init__(self):
 
     def append(
             self,
-            optimizer_result: 'pypesto.OptimizerResult'
+            optimizer_result: 'optimize.OptimizerResult',  # noqa: F821
```

Comment:
I find those `noqa`s pretty ugly. Do we still need that even when using `'pypesto.optimize.OptimizerResults'`?

Reply:
to use `pypesto.optimize.OptimizerResult`, we need to `import pypesto.optimize`. But that module uses the result module -> cyclic dependency ...

Reply:
`from __future__ import annotations`, in case we use python37 as minimum version

Reply:
thanks, I will try that. currently 3.6 but happy to increase to 3.7